includes/functions.php

```diff
@@ -115,7 +115,21 @@ function get_post_embed_html( $post = null, $width, $height ) {
 
 	$embed_url = get_post_embed_url( $post );
 
-	$output = "<script type='text/javascript'>\n";
+	// Get the post excerpt for the given post. This requires setting up the
+	// global data.
+	$previous_post = $GLOBALS['post'];
+	$GLOBALS['post'] = $post;
+	setup_postdata( $GLOBALS['post'] );
+
+	$output = '<noscript>' . get_the_excerpt_embed() . "</noscript>\n";
+
+	// Restore to the previous post.
+	$GLOBALS['post'] = $previous_post;
+	if ( $previous_post ) {
+		setup_postdata( $GLOBALS['post'] );
+	}
```

Comment:
This whole bit is super nasty, but unfortunately the only real way to do it. `wp_reset_postdata` doesn't work inside nested loops, and kind of :hankey:s all over the globals.